layouts/shortcodes/list-examples.html

```diff
@@ -1,25 +1,139 @@
+{{- $filters := split (.Get "filters") "," -}}
+
 {{- $target := printf "%sexamples/" .Page.FirstSection.RelPermalink -}}
 {{- $pages := (where .Site.Pages "Section" .Page.Section) -}}
 
 {{- with .Get "directory" -}}
-  {{- $target = . -}}
+{{- $target = . -}}
 {{- end -}}
 
+<!-- Build out the filters dynamically -->
+{{ if ne (index $filters 0) "" }}
+<div class="selectorDropdowns">
+  {{- range $index, $filter := $filters -}}
+
+  {{- $optionsArray := slice -}}
+  {{- range $index, $page := $pages.ByWeight -}}
+  {{- with (index $page.Params $filter) }}
+  {{- $optionsArray = $optionsArray | append . -}}
+  {{- end -}}
+  {{- end -}}
+
+  {{- $optionsArray = sort (uniq $optionsArray) -}}
 
+  <div>
+    <label for="{{$filter}}">{{$filter | title}}</label>
+    <select class="selectorFilter" name="data-{{$filter}}" id="{{$filter}}">
+      <option value="all">Select...</option>
+      {{- range $optionsArray -}}
+      <option value="{{.}}">{{. | title}}</option>
+      {{- end -}}
+    </select>
+  </div>
+  {{- end -}}
+</div>
+{{ end }}
+
+<!-- Add list of examples, with optional filterable properties -->
 <ul class="DocsCodeExamplesOverview" style="list-style: none">
-  {{- range $pages.ByWeight -}}
-    {{- $x := .RelPermalink -}}
-    {{- $isExample := (strings.HasPrefix $x $target) -}}
-    {{- if and (ne $x $target) $isExample -}}
-    <li class="DocsCodeExamplesOverview--example">
-      <div class="DocsCodeExamplesOverview--example-title">
-        {{- .Page.RenderString (printf "[%s](%s)" .Title .RelPermalink) -}}
-      </div>
-
-      <div class="DocsCodeExamplesOverview--example-description">
-        <p>{{ .Params.summary }}</p>
-      </div>
-    </li>
-    {{- end -}}
+  {{- range $index, $page := $pages.ByWeight -}}
+  {{- $x := $page.RelPermalink -}}
+  {{- $isExample := (strings.HasPrefix $x $target) -}}
+  {{- if and (ne $x $target) $isExample -}}
+  <li class="DocsCodeExamplesOverview--example" {{ if ne (index $filters 0) "" }} {{ range $index2, $filter :=$filters
+    }} {{ with (index $page.Params $filter) }} data-{{$filter}}="{{ delimit (.) " |" }}" {{ end }} {{ end }} {{ end }}>
```

Comment:
```suggestion
    }} {{ with (index $page.Params $filter) }} data-{{$filter}}="{{ delimit (.) "|" }}" {{ end }} {{ end }} {{ end }}>
```
This is causing filtering issues, since `"JavaScript"` != `"JavaScript "` (with a space at the end) when doing the filtering.